azext_edge/edge/providers/support/base.py

```diff
@@ -494,37 +470,45 @@ def default_bundle_name(system_name: str) -> str:
     return f"support_bundle_{timestamp}_{system_name}.zip"
 
 
-def _capture_init_container_logs(
+def _capture_pod_container_logs(
+    pod_containers: List[V1Container],
     pod_name: str,
     pod_namespace: str,
-    pod_spec: V1PodSpec,
     resource_api: EdgeResourceApi,
     v1_api: client.CoreV1Api,
+    capture_previous_logs: bool = True,
     since_seconds: int = 60 * 60 * 24,
 ) -> List[dict]:
 
     processed = []
-    pod_init_containers: List[V1Container] = pod_spec.init_containers or []
-
-    for init_container in pod_init_containers:
-        try:
-            logger.debug(f"Reading init log from pod {pod_name} init container {init_container.name}")
-            log: str = v1_api.read_namespaced_pod_log(
-                name=pod_name,
-                namespace=pod_namespace,
-                since_seconds=since_seconds,
-                container=init_container.name,
-                previous=False,
-            )
-            processed.append(
-                {
-                    "data": log,
-                    "zinfo": (
-                        f"{pod_namespace}/{resource_api.moniker}/pod.{pod_name}.{init_container.name}.init.log"
-                    ),
-                }
-            )
-        except ApiException as e:
-            logger.debug(e.body)
+    capture_previous_log_runs = [False]
+
+    if capture_previous_logs:
+        capture_previous_log_runs.append(True)
+
+    for container in pod_containers:
+        for capture_previous in capture_previous_log_runs:
+            try:
+                logger_debug_previous = "previous run " if capture_previous else ""
+                logger.debug(f"Reading {logger_debug_previous} log from pod {pod_name} container {container.name}")
```

Comment:
There is going to be an extra space added to the debug message because a space was added between
"{logger_debug_previous}" and "log".

Reply:
updated